earth_enterprise/src/fusion/autoingest/sysman/InNodeVertexIndexMap.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 The Open GEE Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IN_NODE_VERTEX_INDEX_MAP_H
+#define IN_NODE_VERTEX_INDEX_MAP_H
+
+#include "DependentStateTree.h"
+
+// The depth_first_search function needs a way to map vertices to indexes. We
+// store a unique index inside each vertex; the code below provides a way for
+// boost to access them. These must be defined before including
+// depth_first_search.hpp.
+template <class Graph>
+class InNodeVertexIndexMap {
+  public:
+    typedef boost::readable_property_map_tag category;
+    typedef size_t value_type;
+    typedef value_type reference;
```

Comment:
I'm not seeing 'reference' used anywhere, at least in this file. Is this typedef needed by boost?

Reply:
Yes, it's used by boost. If I remove any of the typedefs I get a compile error.